source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/user-dashboard/index.tsx

```diff
@@ -0,0 +1,185 @@
+    );
+
+    await updateStore(t => t.replaceRelatedRecords(
+      { type: TYPE_USER, id: user.id },
+      'groups',
+      [{ type: TYPE_GROUP, id: 'group-1' }, { type: TYPE_GROUP, id: 'group-2' }]
+    ), { devOnly: true });
+
+    await updateStore(t => t.replaceRelatedRecord(
+      { type: TYPE_USER, id: user.id },
+      'role',
+      { type: TYPE_ROLE, id: 'role-1' }
+    ), { devOnly: true });
+
+  }
+
+  componentWillMount() {
+    this.generateRandomGroups();
+    this.generateRandomRole();
+    this.generateRandomUsers();
+  }
+
+
+  render() {
+
+    const { users, groups } = this.props;
+
+    console.log(users);
+
+    return (
+      <div className='ui container'>
+        <div className='flex justify-content-space-between'>
+          <h1 className='page-heading'>Manage Users</h1>
+          <div className='flex align-items-center'>
+            <div className='ui left icon input'>
+              <input type="text" placeholder="Search..."/>
+                <i className='search icon'/>
+            </div>
+          </div>
+        </div>
+        {
+          users ?
+          <Table>
+            <Table.Header>
+              <Table.Row>
+                <Table.HeaderCell>Name</Table.HeaderCell>
+                <Table.HeaderCell>Role</Table.HeaderCell>
+                <Table.HeaderCell style={{width: '25%'}}>Groups</Table.HeaderCell>
+                <Table.HeaderCell>Disabled</Table.HeaderCell>
+              </Table.Row>
+            </Table.Header>
+            <Table.Body>
+            {
+              users.map((user, index) => {
+                const { firstName, lastName } = user.attributes;
+
+                //const role = user.relationships['role'].data || {};
+
+                return (
+                  <Table.Row key={index}>
+                    <Table.Cell>
+                      <Link to={'/projects'}>{`${firstName} ${lastName}`}</Link>
+                    </Table.Cell>
+                    <Table.Cell>
+                      {`Builder`}
+                    </Table.Cell>
+                    <Table.Cell>
+                      <GroupDropdown
+                        items={[{id: 'g1', value: 'North America'},{id: 'g2', value: 'South America'}, {id:'g3', value: 'East Asia'}]}
+                        selected={[]}/>
+                    </Table.Cell>
+                    <Table.Cell>
+                      <span></span>
+                    </Table.Cell>
+                  </Table.Row>
+                )
+              })
+            }
+            </Table.Body>
+          </Table> :
+          <div className='flex justify-content-center'>
+            <h3>No user found.</h3>
+          </div>
+        }
+      </div>
+    );
+
+  }
+
+}
+
+const mapRecordsToProps = (ownProps) => {
+  return {
+    users: q => q.findRecords(TYPE_USER),
+    groups: q => q.findRecords(TYPE_GROUP)
+  };
+};
+
+export default compose(
+  requireAuth,
+  withData(mapRecordsToProps)
+)(UserDashboard);
```

Comment:
for bigger components like this that are heavily reliant on data, I think some more composition could be handy -- checkout what I'm doin in this PR: https://github.com/sillsdev/appbuilder-portal/pull/30/files#diff-9a0d98c9f9110d6ead7b32171443db85R11